core/mempool/src/tx_cache.rs

```diff
@@ -203,12 +204,16 @@ impl TxCache {
                     continue;
                 }
                 // After previous filter, tx are valid and should cache in temp_queue.
-                queue_role
+                if let Err(e) = queue_role
                     .candidate
                     .push(Arc::<TxWrapper>::clone(&shared_tx))
-                    .map_err(|_| MemPoolError::InsertCandidate {
-                        len: queue_role.candidate.len(),
-                    })?;
+                {
```

Comment:
If the queue is already full, what is the point of continuing loop?

Reply:
When this happens, there are only a few transactions remaining, so even if the loop goes down, it will basically not consume performance. However, if you use the jump-out method, you need a more complicated design, which is not necessary.

Reply:
But I should remove these pop-out txs from the map to avoid a memory leak.